### PR TITLE
improvement: Added ApplePay settings for demo App

### PR DESCRIPTION
**Pull request by neelSharma12**
### Chore
<chore>

-  Added ApplePay settings.

Simulator Screen Shot - iPhone 14 Pro - 2023-08-14 at 10 16 13

### CODE REVIEW

File: Demo/Common/Configuration/ApplePaySettingsView.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+//
+
+import SwiftUI
+
+@available(iOS 13.0.0, *)
+internal struct ApplePaySettingsView: View {
+    @ObservedObject internal var viewModel: ConfigurationViewModel
+
+    private enum ConfigurationSection: String, CaseIterable {
+        case merchantIdentifier = "{YOUR_APPLE_PAY_MERCHANT_IDENTIFIER}"
```
Comment (descorp) [inline Demo/Common/Configuration/ApplePaySettingsView.swift]: Maybe use `merchant.com.domainname.appname` like on Apple Web Site?

Screenshot 2023-08-14 at 13 26 16

Comment (descorp) [reply Demo/Common/Configuration/ApplePaySettingsView.swift]: Does it have to be `case` ? can we just use `static let` ?

Comment (neelSharma12) [reply Demo/Common/Configuration/ApplePaySettingsView.swift]: yes we can add static let :) I will change it. :)

Comment (neelSharma12) [reply Demo/Common/Configuration/ApplePaySettingsView.swift]: c48ebc09d2550f9e7e445b4a42111155958e0b9a

**Review by descorp — DISMISSED** [review]

File: Demo/Common/Configuration/ApplePaySettingsView.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+//
+
+import SwiftUI
+
+@available(iOS 13.0.0, *)
+internal struct ApplePaySettingsView: View {
+    @ObservedObject internal var viewModel: ConfigurationViewModel
+
+    private enum ConfigurationSection {
+        static let merchantIdentifier = "{ merchant.com.domainname.appname }"
```
Comment (goergisn) [inline Demo/Common/Configuration/ApplePaySettingsView.swift]: Maybe
```suggestion
        static let merchantIdentifier = "Your Pay Merchant ID"
```

File: Demo/Common/Configuration/ConfigurationView.swift [hunk 3]
```diff
@@ -51,6 +52,7 @@ internal struct ConfigurationView: View {
                 wrapInSection(view: paymentSection, section: .payment)
                 wrapInSection(view: cardComponentSection, section: .cardComponent)
```
Comment (goergisn) [inline Demo/Common/Configuration/ConfigurationView.swift]: What about having only 2 sections:
- `DropIn`
- `Payment Methods` or `Components`

File: Demo/Common/Configuration/DropInSettingsView.swift [hunk 4]
```diff
@@ -31,7 +31,8 @@ internal struct DropInSettingsView: View {
                     }
                 }
             }
+            .navigationBarTitle("")
+            .navigationBarHidden(true)
```
Comment (goergisn) [inline Demo/Common/Configuration/DropInSettingsView.swift:35]: Does this actually hide the navigation bar? How would a user be able to go back?

File: Demo/Configuration.swift [hunk 5]
```diff
@@ -207,6 +215,22 @@ internal struct DemoAppSettings: Codable {
 
         return dropInConfig
     }
+
+    internal var applePaySettings: ApplePayComponent.Configuration? {
```
Comment (goergisn) [inline Demo/Configuration.swift:219]: This does not seem to be used 🤔

File: Demo/Common/Configuration/ApplePaySettingsView.swift (hunk 2 above)
Comment (neelSharma12) [reply Demo/Common/Configuration/ApplePaySettingsView.swift]: This change was suggested by @descorp :)

File: Demo/Common/Configuration/ConfigurationView.swift (hunk 3 above)
Comment (neelSharma12) [reply Demo/Common/Configuration/ConfigurationView.swift]: Yes. we can do this :) thanks for the suggestion :)

File: Demo/Common/Configuration/DropInSettingsView.swift (hunk 4 above)
Comment (neelSharma12) [reply Demo/Common/Configuration/DropInSettingsView.swift:35]: I saw some extra space on top of the list when I tested the settings section on device and that was fixed by using this. This does not hide the navigation bar.

File: Demo/Configuration.swift (hunk 5 above)
Comment (neelSharma12) [reply Demo/Configuration.swift:219]: Yes, I corrected it.

**Review by erenbesel — APPROVED** [review]

**Review by goergisn — APPROVED** [review]

**Pull request merged**